KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift

```diff
@@ -279,7 +285,9 @@ final class KarhooLoyaltyPresenter: LoyaltyPresenter {
             return
         }
         
-        let request = LoyaltyPreAuth(identifier: viewModel.loyaltyId, currency: viewModel.currency, points: viewModel.burnAmount, flexpay: false, membership: "")
+        let flexPay = currentMode == .earn // This will be adapted when we integrate flexPay
+        let points = currentMode == .burn ? viewModel.burnAmount : 0 // https://karhoo.atlassian.net/wiki/spaces/PAY/pages/4343201851/Loyalty
```

Comment:
We could add a function getLoyaltyBurnPoints and extract this logic

Reply:
done